Add support for BigQuery `LOAD DATA` statement

VMois opened the pull request:
### Brief summary of the change made
<!--Please include `fixes #XXXX` to automatically close any corresponding issue when the pull request is merged. Alternatively if not fully closed you can say `makes progress on #XXXX`.-->

Adding support to LOAD DATA statement in BigQuery dialect.

closes #6589 

### Are there any other side effects of this change that we should be aware of?

Not that I am aware of. BigQuery parser tests passed successfully. Seems like the LOAD DATA statement did not break any other BigQuery statements.

### Pull Request checklist

- [x] `.sql`/`.yml` parser test cases in `test/fixtures/dialects`. Used `tox` as described in the Contributing guide.
- [x] No follow-up issues are necessary.

VMois commented on test/fixtures/dialects/bigquery/load_data_statement.sql:
```diff
@@ -0,0 +1,106 @@
+LOAD DATA INTO mydataset.table1
```
*note:* I copied all examples from [BQ docs website](https://cloud.google.com/bigquery/docs/reference/standard-sql/load-statements#examples).

VMois replied:
*question:* do we care here to preserve the dialect logic? The example below will parse but fail in BigQuery.

```sql
-- This query returns an error.
LOAD DATA INTO mydataset.table1
  (
    x INT64, -- column_list is given but the partition column list is missing
    y STRING
  )
  FROM FILES(
    format='AVRO',
    uris = ['gs://bucket/path/*'],
    hive_partition_uri_prefix='gs://bucket/path'
  )
  WITH PARTITION COLUMNS
```

I assume as a parser we do not care about dialect logic. But want to make sure.

VMois commented:
*question:* does adding support to a statement require updating any docs?

VMois commented on src/sqlfluff/dialects/dialect_bigquery.py:
```diff
@@ -2683,6 +2684,81 @@ class ExportStatementSegment(BaseSegment):
+        OneOf("INTO", "OVERWRITE"),
+        Sequence(OneOf("TEMP", "TEMPORARY", optional=True), "TABLE", optional=True),
+        Ref("TableReferenceSegment"),
+        Sequence(
+            Bracketed(
+                Delimited(
+                    OneOf(
+                        Ref("ColumnDefinitionSegment"),
+                        Ref("TableConstraintSegment"),
+                    ),
+                    allow_trailing=True,
+                )
+            ),
+            optional=True,
+        ),
+        Sequence(
```
*question:* This snippet is taken from `CreateTableStatementSegment`. Not sure if there is a need to consolidate it into a separate segment or if duplication is acceptable.

I would keep it as it is. Open to suggestions otherwise.

keraion commented:
> _question:_ does adding support to a statement require updating any docs?

Nope, but including the dialect source documentation in the code like you did is always super helpful.

VMois commented on src/sqlfluff/dialects/dialect_bigquery.py:
```diff
@@ -2683,6 +2684,81 @@ class ExportStatementSegment(BaseSegment):
+        "LOAD",
+        "DATA",
+        OneOf("INTO", "OVERWRITE"),
+        Sequence(OneOf("TEMP", "TEMPORARY"), "TABLE", optional=True),
```
Nice. Will update code a bit later today.

VMois commented on src/sqlfluff/dialects/dialect_bigquery.py:
```diff
@@ -2683,6 +2684,81 @@ class ExportStatementSegment(BaseSegment):
+        Ref("TableReferenceSegment"),
+        Sequence(
+            Bracketed(
+                Delimited(
+                    OneOf(
+                        Ref("ColumnDefinitionSegment"),
+                        Ref("TableConstraintSegment"),
+                    ),
+                    allow_trailing=True,
+                )
```
I think I tried without it and it failed some parsing. Let me double check it here a bit later today.

VMois replied:
After removing "OneOf", all examples are parsable. Probably I mixed with some other error.

VMois commented:
Addressed all *nits*. Rebased against main to get latest changes. All seems good. Not sure what is the merge process here. Will leave it up to the maintainers.

Thank you very much for the quick review. SQLFluff is great!
